### PR TITLE
Switch date picker to using a custom Dialog

### PR DESCRIPTION
**Pull request by abarth**
Rather than removing all the padding from an AlertDialog, we now just use
Dialog directly for the date picker.

### CODE REVIEW

**Comment by abarth** [comment]
@dragostis @Hixie

**Comment by abarth** [comment]
Also added another commit that supports landscape date pickers.

File: packages/flutter/lib/src/material/date_picker.dart [hunk 1]
```diff
@@ -580,3 +512,185 @@ class _YearPickerState extends State<YearPicker> {
...
+                    )
+                  ]
+                )
+              );
+          }
+          return null;
+        }
+      )
+
+
```
Comment (Hixie) [inline packages/flutter/lib/src/material/date_picker.dart]: maybe a little too much whitespace here

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/lib/src/material/date_picker.dart [hunk 2]
```diff
@@ -580,3 +512,185 @@ class _YearPickerState extends State<YearPicker> {
...
+          );
+          assert(orientation != null);
+          switch (orientation) {
+            case Orientation.portrait:
+              return new SizedBox(
+                width: _kMonthPickerPortraitWidth,
+                child: new Column(
+                  mainAxisSize: MainAxisSize.min,
+                  crossAxisAlignment: CrossAxisAlignment.stretch,
+                  children: <Widget>[ header, picker, actions ]
```
Comment (dragostis) [inline packages/flutter/lib/src/material/date_picker.dart]: Should there be space around the array brackets?

**Review by dragostis — APPROVED** [review]
LGTM.

File: packages/flutter/lib/src/material/date_picker.dart (hunk 2 above)
Comment (abarth) [inline packages/flutter/lib/src/material/date_picker.dart]: Looks like we're slighly inconsistent.  I'll remove it.

**Pull request merged**